tool/site/client/src/pages/configPage.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from "react"
+import {useContext, useEffect, useState} from "react"
+import {Label, Button, Fieldset, Form, Checkboxes, Input, CrossIcon, TickIcon} from "nhsuk-react-components"
+import {AppContext} from "../index"
+import ButtonList from "../components/buttonList"
+import {Field, Formik} from "formik"
+import {axiosInstance} from "../requests/axiosInstance"
+import BackButton from "../components/backButton"
+
+interface ConfigFormValues {
+  useSigningMock: boolean
+  signingPrNumber: string
+}
+
+const ConfigPage: React.FC = () => {
+  const {baseUrl} = useContext(AppContext)
+  const [configFormValues, setConfigFormValues] = useState<ConfigFormValues>()
+  const [configUpdateSuccess, setConfigUpdateSuccess] = useState(undefined)
+  const initialValues = {useSigningMock: false, signingPrNumber: undefined}
+
+  useEffect(() => {
+    if (configFormValues) {
+      (async () => {
+        const success = await updateConfig(baseUrl, configFormValues)
+        setConfigUpdateSuccess(success)
+      })()
+    }
+  }, [baseUrl, configFormValues, setConfigUpdateSuccess])
```

Comment:
I'm a little confused why we are using a `useEffect` here? 

If I've understood the logic when we click `Save` that calls `setConfigFormValues` which triggers this useEffect, this does the updated if the `configFormValues` are correct.

Could we not have this logic inside the useEffect being triggered directly by the buttons `onSubmit` handler?